src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -1527,11 +1527,15 @@ See https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-change
             .collect::<CargoResult<Vec<_>>>()?
     };
 
+    // Environment variables entered in the build script as part of the fingerprint
+    let rustflags = build_runner.bcx.rustflags_args(unit).to_vec();
```

Comment:
If I'm reading this correctly, we are blindly adding the target rustflags to the fingerprint for build script units.  This will cause build scripts to rebuild/re-run more often than is needed.

We should be limiting this to when the build script cares about it.